libblkid/src/superblocks/iso9660.c

```diff
@@ -44,10 +47,10 @@ struct iso_volume_descriptor {
 	unsigned char	space_size[8];
 	unsigned char	escape_sequences[8];
 	unsigned char  unused2[94];
-	unsigned char  volume_set_id[128];
-	unsigned char  publisher_id[128];
-	unsigned char  data_preparer_id[128];
-	unsigned char  application_id[128];
+	unsigned char  volume_set_id[ISO_ID_BUFSIZ];
+	unsigned char  publisher_id[ISO_ID_BUFSIZ];
+	unsigned char  data_preparer_id[ISO_ID_BUFSIZ];
+	unsigned char  application_id[ISO_ID_BUFSIZ];
```

Comment:
I do not think that it is correct to use `ISO_ID_BUFSIZ` macro in fixed structure format. Per ISO 9660 spec application id is always 128 bytes long. But `ISO_ID_BUFSIZ` is defined as _maximum field size passed to merge_utf16be_ascii()_ which sounds like something which would be increased in future once some other bigger buffer is passed to `merge_utf16be_ascii`. Also you are using same merge_utf16be_ascii buffer for `volume_id` which (obviously) does not use `ISO_ID_BUFSIZ`.

Reply:
Does the `sizeof()` really help? in case of changes somebody first has to resolve the `sizeof()` to figure out if `ISO_MAX_FIELDSIZ` has to be adapted.

What about:

```c
struct iso_volume_descriptor {
  ...
  unsigned char volume_set_id [128];
}

/* maximal field size used anywhere in ISO; update if necessary */
#define ISO_MAX_FIELDSIZ  128

...
unsigned char buf[ISO_MAX_FIELDSIZ  * 5 / 2];
```

Reply:
Here we go :-)